### PR TITLE
IT-4998 - feature icon

### PR DESCRIPTION
**Pull request by damianmef**

### CODE REVIEW

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		},
+	},
+	data() {
+		return {
+			FEATURE_ICON_COLOR: Object.freeze(FEATURE_ICON_COLOR),
+			FEATURE_ICON_SIZES: Object.freeze(FEATURE_ICON_SIZES),
+		};
+	},
+	computed: {
+		sizeClassName() {
```
Comment (kvas-damian) [inline lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:162]: not used

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		color: {
+			type: String,
+			required: true,
+			validator(color) {
+				return Object.values(FEATURE_ICON_COLOR).includes(color);
+			},
+		},
+		doubleBackground: {
+			type: Boolean,
+			default: true,
```
Comment (kvas-damian) [inline lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:152]: according to agreement with UX, it should be `false`

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+		},
+		size: {
+			type: String,
+			default: FEATURE_ICON_SIZES.MEDIUM,
+			validator: (value: string) => Object.values(FEATURE_ICON_SIZES).includes(value),
+		},
+		color: {
+			type: String,
+			required: true,
+			validator(color) {
```
Comment (kvas-damian) [inline lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:146]:
```suggestion
			validator(color: string) {
```

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+		'icon': $color-warning-icon,
+	),
+	'info': (
+		'background': $color-info-background-medium,
+		'border': $color-info-background,
+		'icon': $color-info-icon,
+	),
+);
+
+$feature-icon-padding-large: 10px;
```
Comment (kvas-damian) [inline lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:66]: I'm little bit worried that we have to create custom padding for this size. Maybe UX intention was to have fixes size for this component and not use padding at all? Can we confirm what is going on here?

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 5]
```diff
@@ -0,0 +1,182 @@
+<template>
+	<div
+		class="featureIcon"
+		:class="{
+			'-doubleBackground': doubleBackground,
+			'-danger': color === FEATURE_ICON_COLOR.DANGER,
+			'-info': color === FEATURE_ICON_COLOR.INFO,
+			'-neutral': color === FEATURE_ICON_COLOR.NEUTRAL,
+			'-neutral-weak': color === FEATURE_ICON_COLOR.NEUTRAL_WEAK,
```
[the remark this answers is not in the document]
Comment (damianmef) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue]: Updated.

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue (hunk 1 above)
Comment (damianmef) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue]: Removed.

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue (hunk 2 above)
Comment (damianmef) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue]: Fixed.

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue (hunk 3 above)
Comment (damianmef) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue]: Updated.

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue (hunk 4 above)
Comment (damianmef) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue]: Code updated, now we base on the const dimensions, not padding.

Comment (kvas-damian) [reply lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:66]: Agreed with UX, we use custom padding here, as it's more flexible when we change Icon sizes in future

File: lib/js/components/Icons/FeatureIcon/FeatureIcon.vue [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+	'info': (
+		'background': $color-info-background-medium,
+		'border': $color-info-background,
+		'icon': $color-info-icon,
+	),
+);
+
+$feature-icon-padding-large: 10px;
+
+@mixin setBorderColor($border) {
```
Comment (kvas-damian) [inline lib/js/components/Icons/FeatureIcon/FeatureIcon.vue:68]: Does it really makes sense to create a mixin for one property?

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**